v3/src/components/case-table/checkbox-cell.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useRef } from "react"
+import { useDataSetContext } from "../../hooks/use-data-set-context"
+import { FValue } from "../../models/formula/formula-types"
+
+export const isBoolean = (value: FValue | undefined) => {
+  let v = value
+  if (typeof v === "string") {
+    v = v.toLowerCase()
+  }
+  return v === undefined || ["", "false", "true", false, true, null].includes(v as string | boolean | null)
+}
+
+interface ICheckboxCellProps {
+  rowId: string
+  attrId: string
+}
+
+export default function CheckboxCell ({ rowId, attrId }: ICheckboxCellProps) {
+  const data = useDataSetContext()
+  // We need checkRef to show indeterminate state
+  const checkRef = useRef<HTMLInputElement>(null)
+  const cellValue = data?.getValue(rowId, attrId)
+
+  useEffect(() => {
+    if (checkRef.current) {
+      const isBool = isBoolean(cellValue)
+      const isTrue = typeof cellValue === "string" && cellValue.toLowerCase() === "true"
```

Comment:
```suggestion
      const isTrue = cellValue === true || (typeof cellValue === "string" && cellValue.toLowerCase() === "true")
```
`data?.getValue()` can return `boolean` values.